### ISSUE TITLE
discord.js and voice's types are incompatible 

### ISSUE THREAD

**Issue by Lechixy** [discordjs/discord.js#7221]
### Which package is this bug report for?

discord.js

### Issue description

![image](https://user-images.githubusercontent.com/51778131/148666330-df7214ce-f0a6-4e39-a7a0-0a2db9dfbd34.png)

as you can see typescript showing an error type "InternalDiscordGatewayAdapterCreator" are not assignable to 
"DiscordGatewayAdapterCreator"

### Package version

13.15.1

### Node.js version

Node: v16.6.1 and Typescript: v4.5.4

### Priority this issue should have

Medium (should be fixed soon)

### Which partials do you have configured?

USER, CHANNEL, GUILD_MEMBER, MESSAGE, REACTION

### Which gateway intents are you subscribing to?

GUILDS, GUILD_MEMBERS, GUILD_VOICE_STATES, GUILD_PRESENCES, GUILD_MESSAGES, GUILD_MESSAGE_REACTIONS

**Comment by VlaDexa** [comment]
Reproduction

```ts
import discord from 'discord.js'
import {Intents} from 'discord.js'
import {joinVoiceChannel} from '@discordjs/voice'
const client = new discord.Client({ intents : [Intents.FLAGS.GUILDS, Intents.FLAGS.GUILD_MESSAGES, Intents.FLAGS.GUILD_VOICE_STATES, Intents.FLAGS.DIRECT_MESSAGES] , partials : ['CHANNEL', 'MESSAGE']})

client.on("messageCreate", async (message) => {
    if (message.guild && message.member){
		if (!message.member.voice?.channel) return;
        const connection = joinVoiceChannel({
            channelId : message.member.voice.channel.id,
            guildId : message.guild.id,
            adapterCreator: message.guild.voiceAdapterCreator
        })
    }
})
```

**Comment by Lechixy** [comment]
it fixed in 1.13.6, thanks
